upgrades/upgrade_testing_framework/core/expectation.py

```diff
@@ -0,0 +1,96 @@
+from pathlib import Path
+import json
+from typing import List
+import operator
+
+from upgrade_testing_framework.core.versions_engine import EngineVersion
+
+class KnowledgeBaseDoesntExistException(Exception):
```

Comment:
nit: It might be clearer here to rename this as KnowledgeBaseDirectoryDoesntExistException